src/backend/renderer/sync/mod.rs

```diff
@@ -0,0 +1,91 @@
+//! Helper for synchronizing rendering operations
+use std::{os::unix::io::OwnedFd, sync::Arc};
+
+use downcast_rs::{impl_downcast, Downcast};
+
+#[cfg(feature = "backend_egl")]
+mod egl;
+
+/// A fence that will be signaled in finite time
+pub trait Fence: std::fmt::Debug + Send + Sync + Downcast {
+    /// Queries the state of the fence
+    fn is_signaled(&self) -> bool;
+
+    /// Blocks the current thread until the fence is signaled.
+    fn wait(&self);
+
+    /// Returns whether this fence has a native handle
+    fn is_native(&self) -> bool;
+
+    /// Get the native fence fd if available
+    fn native(&self) -> Option<OwnedFd>;
+}
+impl_downcast!(Fence);
+
+/// A sync point the will be signaled in finite time
+#[derive(Debug, Clone)]
+#[must_use]
```

Comment:
We can put a message here if useful that the compiler will add to diagnostic's:
```rust
#[must_use = "message here"]
```

Reply:
added a message